pipeline/DESIGN.md

```diff
@@ -0,0 +1,212 @@
+| data        | bytes             | DAG-JSON encoding of the event payload                                                          |
+| previous    | list(bytes)       | Ordered list of CID previous to this event. Meaning of the order is stream type dependent       |
+
+
+### Transformation
+
+Raw events are transformed into a flattened structure where stream dimensions and conclusions about data are added to the data.
+
+Conclusions include:
+
+* The event has a valid signature
+* The dimensions and controller of the event
+
+This table joins the raw_events, chain_proofs, and streams tables in order to make the conclusions about the raw events.
+
+## model_schemas
+
+The model_schemas table contains a row for each model and contains the complete resolved schema of the model.
+
+TBD how this table is populated and its schema.
+This table may be able to be combined with the streams table. Should we?
+
+## event_states
+
+The event_states table contains a row for each event in a stream and the state of the document at that point in the stream.
+
+### Features
+
+* Access to the full history of states for a stream
+* Validation of the model schema
+
+### Schema
+
+| Column      | Type              | Description                                                                                     |
+| ------      | ----              | -----------                                                                                     |
+| index       | u64               | Order of this event. Index is always greater than the index of any previous event in the stream |
+| stream_cid  | bytes             | Cid of the stream                                                                               |
+| stream_type | u8                | Type of the stream, see [stream type values](#stream-types)                                     |
+| controller  | string            | Controller of the stream                                                                        |
+| dimensions  | map(string,bytes) | Set of key values dimension pairs of the stream                                                 |
+| event_cid   | bytes             | Cid of the event                                                                                |
+| event_type  | u8                | Type of the event, see [event type values](#event-types)                                        |
+| state       | bytes             | JSON encoding of the event state                                                                |
+
+### Transformation
+
+This table computes the aggregated state for each conclusion event.
+Additionally it validates the aggregated state matches the model schema of the stream.
+
+## stream_tips
+
+The stream_states table contains a row for head of each stream representing the canonical state of the stream tips.
+
+### Features
+
+* Access to the multiple tips of streams for users building their on conflict resolution
+
+### Schema
+
```

Comment:
should we add event height to the schema here?  Seems like it will be needed in the next pipeline phase to do conflict resolution